src/wolfgame.py

```diff
@@ -1720,73 +1730,68 @@ def stats(cli, nick, chan, rest):
     reply(cli, nick, chan, stats_mssg)
 
 @handle_error
-def hurry_up(cli, gameid, change):
+def hurry_up(gameid, change):
     if var.PHASE != "day": return
     if gameid:
         if gameid != var.DAY_ID:
             return
 
-    chan = botconfig.CHANNEL
-
     if not change:
         event = Event("daylight_warning", {"message": "daylight_warning"})
         event.dispatch(var)
-        cli.msg(chan, messages[event.data["message"]])
+        channels.Main.send(messages[event.data["message"]])
         return
 
     var.DAY_ID = 0
 
-    pl = set(list_players()) - (var.WOUNDED | var.CONSECRATING)
+    pl = set(get_players()) - (var.WOUNDED | var.CONSECRATING)
     evt = Event("get_voters", {"voters": pl})
     evt.dispatch(var)
     pl = evt.data["voters"]
+    not_lynching = set(var.NO_LYNCH)
 
     avail = len(pl)
     votesneeded = avail // 2 + 1
-    not_lynching = len(var.NO_LYNCH)
 
-    avail = len(pl)
-    votesneeded = avail // 2 + 1
-    not_lynching = set(var.NO_LYNCH)
-    votelist = copy.deepcopy(var.VOTES)
-
-    # Note: this event can be differentiated between regular chk_decision
-    # by checking evt.params.timeout.
-    event = Event("chk_decision", {
-        "not_lynching": not_lynching,
-        "votelist": votelist,
-        "numvotes": {}, # filled as part of a priority 1 event
-        "weights": {}, # filled as part of a priority 1 event
-        "transition_night": transition_night
-        }, voters=pl, timeout=True)
-    if not event.dispatch(cli, var, ""):
-        return
-    not_lynching = event.data["not_lynching"]
-    votelist = event.data["votelist"]
-    numvotes = event.data["numvotes"]
-
-    found_dup = False
-    maxfound = (0, "")
-    for votee, voters in votelist.items():
-        if numvotes[votee] > maxfound[0]:
-            maxfound = (numvotes[votee], votee)
-            found_dup = False
-        elif numvotes[votee] == maxfound[0]:
-            found_dup = True
+    with copy.deepcopy(var.VOTES) as votelist:
+        # Note: this event can be differentiated between regular chk_decision
+        # by checking evt.params.timeout.
+        event = Event("chk_decision", {
+            "not_lynching": not_lynching,
+            "votelist": votelist,
+            "numvotes": {}, # filled as part of a priority 1 event
+            "weights": {}, # filled as part of a priority 1 event
+            "transition_night": transition_night
+            }, voters=pl, timeout=True)
+        if not event.dispatch(var, None):
+            return
+        not_lynching = event.data["not_lynching"]
+        votelist = event.data["votelist"]
```

Comment:
Does this work in python? Reassigning the thing you're context-managing inside of the context manager seems like something that shouldn't happen. (I know in C# it throws an exception if you try, idk what python's behavior is)

Reply:
It works no problem (was tested), and is intended as it's the variable we should be clearing at the end. Though to be honest, I probably shouldn't re-assign it and just let events mutate.